rapids-cmake/cuda/detail/detect_architectures.cmake

```diff
@@ -64,7 +64,7 @@ int main(int argc, char** argv) {
   return 0;
   }
   ")
-    execute_process(COMMAND ${CMAKE_CUDA_COMPILER} -std=c++11 -o "${eval_exe}" "${eval_file}"
+    execute_process(COMMAND ${CMAKE_CUDA_COMPILER} -std=c++17 -o "${eval_exe}" "${eval_file}"
```

Comment:
Can you share more about what is going wrong here? Do we need added test coverage for the problem? Or at least a sense of why the C++ standard used here affects the correctness? Does the program compile and give wrong results or fail to compile? If you compile this code standalone, what happens?

Reply:
I did a bit of digging.

These are the same issues:
https://github.com/NVIDIA/nccl/issues/1743
https://github.com/PyPO-dev/PyPO/issues/207
https://github.com/Xpra-org/xpra/issues/4424

This was also reported here: https://gitlab.archlinux.org/archlinux/packaging/packages/cuda/-/issues/12
From there, it became an nvbug that was fixed in CUDA 13.0. https://nvbugspro.nvidia.com/bug/5262977

The mitigations are:
- Use `-std=c++14` or higher
- Use GCC < 14.2.0 and/or nvcc >= 13.0 if `-std=c++11` is required (the issue appears to occur for GCC 14.2.1 and higher with CUDA versions prior to 13.0)